embassy-executor/src/raw/mod.rs

```diff
@@ -162,6 +161,12 @@ pub struct TaskStorage<F: Future + 'static> {
     future: UninitCell<F>, // Valid if STATE_SPAWNED
 }
 
+unsafe fn poll_to_despawn(p: TaskRef) {
+    // The task's future has already been dropped, we just mark it as `!SPAWNED`.
+    let this = p.header();
```

Comment:
Yeah this can, thanks. I was considering just storing(or swapping in) SPAWNED|RUN_QUEUED in run_enqueue. If that turns out to be a win, this will re-appear.